### PR TITLE
feat: scheduled reports

### PR DESCRIPTION
**Pull request by ilyozzz**

### CODE REVIEW

File: .github/workflows/aws-staging.yml [hunk 1]
```diff
@@ -110,6 +110,9 @@ jobs:
           export SECRETS_PG_PASS=${{ secrets.PG_PASS }}
           export SECRETS_PG_DB=${{ secrets.PG_DB }}
 
+          export SECRETS_AWS_S3_ACCESS_KEY_ID=${{secrets.AWS_ACCESS_KEY_ID}}
```
Comment (ifokeev) [inline .github/workflows/aws-staging.yml:113]: please add envs to the `docker-compose.stack.yml` too

File: services/actions/src/rpc/sendExplorationScreenshot.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+            return false;
+          }
+
+          resultJsonData = data;
+        } catch (error) {
+          return false;
+        }
+
+        return true;
+      }, { timeout: PUPPETEER_WAITING_TIMEOUT }),
```
Comment (ifokeev) [inline services/actions/src/rpc/sendExplorationScreenshot.js:126]: please create an incapsulated function for the callback

File: services/actions/src/rpc/sendExplorationScreenshot.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+  const height = await page.evaluate((selector) => document.querySelector(selector).scrollHeight, EXPLORATION_DATA_SELECTOR);
+  await page.setViewport({ width, height });
+
+  await page.evaluate(() => {
+    const header = document.querySelector('.ant-layout-header');
+    const footer = document.querySelector('.ant-layout-footer');
+    const sider = document.querySelector('.ant-layout-sider');
+    const alert = document.querySelector('.ant-alert');
+
+    [header, footer, sider, alert].forEach(element => element?.parentNode?.removeChild(element));
```
Comment (ifokeev) [inline services/actions/src/rpc/sendExplorationScreenshot.js:149]: please create an incapsulated function for the callback

File: services/actions/src/utils/s3.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+const {
+  AWS_S3_ENDPOINT,
+  AWS_S3_ACCESS_KEY_ID,
+  AWS_S3_SECRET_ACCESS_KEY,
+  AWS_S3_REGION,
+} = process.env;
+
+// 7 days
+const AWS_S3_PRESIGNED_URL_EXPIRES_IN = 7 * 24 * 60 * 60;
```
Comment (ifokeev) [inline services/actions/src/utils/s3.js:21]: looks like that it should in in env

File: services/client/src/components/ReportModal.js [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+            constraint: 'reports_pkey',
+            update_columns: ['name', 'schedule', 'delivery_config', 'exploration_id']
+          },
+          data: [{
+            id: report.id,
+            name: values.name,
+            schedule: values.schedule,
+            delivery_type: values.delivery_type,
+            delivery_config: values.delivery_config,
+            team_id: currentTeamState?.id,
```
Comment (ifokeev) [inline services/client/src/components/ReportModal.js:98]: to save the author, it should have `user_id` too

File: services/client/src/hooks/useReports.js [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+    }
+  }
+`;
+
+const reportsQuery = `
+  query ($offset: Int, $limit: Int, $where: reports_bool_exp, $order_by: [reports_order_by!]) {
+    reports (offset: $offset, limit: $limit, where: $where, order_by: $order_by) {
+      id
+      name
+      schedule
```
Comment (ifokeev) [inline services/client/src/hooks/useReports.js:37]: needs user_id

**Review by ifokeev — CHANGES_REQUESTED** [review]

File: services/client/src/components/ReportModal.js (hunk 5 above)
Comment (ilyozzz) [reply services/client/src/components/ReportModal.js:98]: it's in hasura presets

File: services/actions/src/rpc/createCronTaskByReport.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    delivery_config: deliveryConfig,
+    schedule,
+    exploration_id: explorationId,
+    name: reportName,
+    id: reportId
+  } = reportPayload || {};
+
+  const cronTaskParams = {
+    type : "create_cron_trigger",
+    args : {
```
Comment (ifokeev) [inline services/actions/src/rpc/createCronTaskByReport.js:25]: style issue, remove space, please


we need eslint

File: services/actions/src/rpc/createCronTaskByReport.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    delivery_type: deliveryType,
+    delivery_config: deliveryConfig,
+    schedule,
+    exploration_id: explorationId,
+    name: reportName,
+    id: reportId
+  } = reportPayload || {};
+
+  const cronTaskParams = {
+    type: "create_cron_trigger",
```
Comment (ifokeev) [inline services/actions/src/rpc/createCronTaskByReport.js:21]: different quotes

File: services/actions/src/rpc/createCronTaskByReport.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        timeout: 60,
+      },
+      include_in_metadata: false,
+      comment: `Cron event for scheduled report id: ${reportId}`
+    }
+  }
+
+  try {
+    // recreate the cron task if fields or exploration has been changed
+    if (operationName === "UPDATE") {
```
Comment (ifokeev) [inline services/actions/src/rpc/createCronTaskByReport.js:44]: different quotes

File: services/actions/src/rpc/deleteCronTaskByReport.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import { fetchMetadataAPI } from '../utils/hasuraMetadataApi';
+import logger from '../utils/logger';
+import apiError from '../utils/apiError';
+
+export default async (session, input) => {
+  const reportId = input?.data?.old?.id;
+
+  const cronTaskParams = {
+    type : "delete_cron_trigger",
```
Comment (ifokeev) [inline services/actions/src/rpc/deleteCronTaskByReport.js:9]: different quotes

**Review by ifokeev — APPROVED** [review]

**Comment by ifokeev** [comment]
closes #30

**Pull request merged**